sdk/storage/azure-storage-blobs/src/blob_client.cpp

```diff
@@ -366,13 +367,13 @@ namespace Azure { namespace Storage { namespace Blobs {
     auto bodyStreamToFile = [](Azure::Core::IO::BodyStream& stream,
                                _internal::FileWriter& fileWriter,
                                int64_t offset,
-                               size_t length,
+                               int64_t length,
                                const Azure::Core::Context& context) {
       constexpr size_t bufferSize = 4 * 1024 * 1024;
       std::vector<uint8_t> buffer(bufferSize);
       while (length > 0)
       {
-        size_t readSize = std::min(bufferSize, length);
+        size_t readSize = std::min(bufferSize, static_cast<size_t>(length));
```

Comment:
This is still problematic. What if `length` is exact `0x100000000`? It will become `0` after casting to `size_`, then it will lead to a infinite loop.

Reply:
`readSize = static_cast<size_t>(std::min<int64_t>(bufferSize, length))`?

Reply:
Yeah, that will work. But I just feel bad about the castings, casting `bufferSize` to `int64_t` first, then casting min of the two back to `size_t`.

What do you think of my [proposal](https://github.com/Azure/azure-sdk-for-cpp/pull/2379#issuecomment-852687282)?

Reply:
Sounds reasonable to me, but I am not deep into the size_t vs int64_t discussion.
I am fine with int64_t. But either way, I here would care more about the tactical stuff which is to make this `std::min` non-problematic, whichever way you chose.